src/API/Query.php

```diff
@@ -190,6 +194,10 @@ private function populateRequest($clientKey)
       }
     }
 
+    if (!empty($this->matchStrategy)) {
+      $request->matchStrategy = self::convertPartialMatchStrategy($this->matchStrategy);
+    }
+
 //        $returnBinary = $this->returnBinary;
```

Comment:
@grrilla Suggestion; A potential positive refactoring would be to remove these comments.

Reply:
These comments refer to a comment higher in this class (where $returnBinary is declared) defining  details of when it should be put back in place. This functionality would be consistent with the Java API, and given the nature of the ticket, I ultimately decided to go with the conservative road. (Leaving it)

Reply:
Sounds good.